### PR TITLE
Support API call cancelation via AbortSignal

### PR DESCRIPTION
**Pull request by robertknight**
This PR adds some general functionality for LMS frontend -> backend API calls which will be used by the JSTOR picker.

1. A `signal` argument to `apiCall`, which can be used to pass an [AbortSignal](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal) to fetch, to enable later cancelation of the request
2. A `urlPath` helper for safely generating API URL Paths that include parameters

### CODE REVIEW

File: lms/static/scripts/frontend_apps/utils/test/api-test.js [hunk 1]
```diff
@@ -272,4 +291,53 @@ describe('api', () => {
     assert.deepEqual(firstCallResult, await fakeResponse.json());
     assert.deepEqual(secondCallResult, await fakeResponse.json());
   });
+
+  it('aborts if fetch is canceled', async () => {
```
Comment (lyzadanger) [inline lms/static/scripts/frontend_apps/utils/test/api-test.js:295]: Though I understand this test's contents, I'm having a little trouble mapping them to this description. 

As I understand from docs, when `AbortController.abort` is called, a `fetch` operation with a corresponding `signal` to that controller will reject with a `DOMException` that looks like the one generated here. As such, this is what an aborted `fetch` would "look like" to a caller. 

OTOH, I'm not clear where the "canceled" piece comes in here. That is, where is the line between "cancel" and "abort"?

File: lms/static/scripts/frontend_apps/utils/api.js [hunk 2]
```diff
@@ -126,3 +131,23 @@ export async function apiCall(options) {
 
   return resultJSON;
 }
+
+/**
+ * Template tag that formats a URL path, ensuring interpolated strings are
```
Comment (lyzadanger) [inline lms/static/scripts/frontend_apps/utils/api.js:136]:
```suggestion
 * Template tag function that formats a URL path, ensuring interpolated strings are
```

Pedantic, but sometimes the difference between "tagged template [literals]" and the associated "tag function" is confusing.

**Review by lyzadanger — APPROVED** [review]
I was unsure about one test, but otherwise this is making sense...

File: lms/static/scripts/frontend_apps/utils/test/api-test.js (hunk 1 above)
Comment (robertknight) [reply lms/static/scripts/frontend_apps/utils/test/api-test.js:295]: > As such, this is what an aborted fetch would "look like" to a caller.

Yes, indeed.

> OTOH, I'm not clear where the "canceled" piece comes in here. That is, where is the line between "cancel" and "abort"?

"Canceled" and "aborted" are synonyms in this context. I'll revise the wording in this description.

**Pull request merged**